en/docs/Learn/APISecurity/OAuth2DeepDive/securing-oauth-token-with-hmac-validation.md

```diff
@@ -15,14 +15,16 @@ If an attacker uses random tokens to send API requests, API Manager will try to
 ![]({{base_path}}/assets/attachments/103334788/103334789.png)
 #### **WSO2 IS Extension - OAuth Token Generator Extension**
 
-Engage the HMAC OAuth handler in order to do the Keyed-Hash Message Authentication Code (HMAC) validation by adding following into &lt;IS\_HOME&gt;/repository/conf/identity/identity.xml
+Engage the HMAC OAuth handler in order to do the Keyed-Hash Message Authentication Code (HMAC) validation by adding following config inti  &lt;APIM_HOME&gt;/repository/conf/deployment.toml
 
-``` java
-    <IdentityOAuthTokenGenerator>com.sample.lahiru.wso2.hmac.oauth</IdentityOAuthTokenGenerator>
+
+```
+[oauth.extensions]
+token_generator="com.sample.lahiru.wso2.hmac.oauth"
 ```
 
 !!! info
-M ore information on developing OAuth token generator extensions [here](https://docs.wso2.com/display/IS530/Extension+Points+for+OAuth#ExtensionPointsforOAuth-OAuthTokenGenerator) . Code for this particular solution can be found in [oauth-hmac-extension](https://github.com/lahirus/oauth-hmac-extension/blob/master/src/main/java/com/sample/lahiru/wso2/hmac/oauth/OAuthHMACTokenIssuer.java) GitHub repository .
+    More information on developing OAuth token generator extensions [here](https://is.docs.wso2.com/en/5.9.0/learn/extension-points-for-oauth/#oauth-token-generator) . Code for this particular solution can be found in [oauth-hmac-extension](https://github.com/lahirus/oauth-hmac-extension/blob/master/src/main/java/com/sample/lahiru/wso2/hmac/oauth/OAuthHMACTokenIssuer.java) GitHub repository .
```

Comment:
More information on developing OAuth token generator extensions can be found here.